### ISSUE TITLE
OrientGraphFactory transaction checking uses extra server connections (v2.0)

### ISSUE THREAD

**Issue by 8forty** [orientechnologies/orientdb#5754]
In our project we have a singleton that manages an OrientGraphFactory and transactions using it.  Inside that code we have a call to OrientGraphFactory.getDatabase().getTransaction().isActive() to check whether the user is executing his query in a transaction or not.  This call appears to create a separate connection every time it's called regardless of whether there's a pool, etc., and can quickly use all available server connections.

Here's a code snippet that demonstrates the problem by creating over 1000 connections to the server:
```java
public class O2ConnectionsTest
{
    private static final String DBURL = "remote:localhost/conntest";
    private static final long RecordCount = 1000;

    public static void main(String[] args) throws InterruptedException
    {
        try
        {
            // reset the db every time
            OServerAdmin admin = new OServerAdmin(DBURL);
            admin.connect("root", "root");
            if (admin.existsDatabase("plocal")) admin.dropDatabase("plocal");
            admin.createDatabase("graph", "plocal");

            final OrientGraphFactory gfactory = new OrientGraphFactory(DBURL, "root", "root");
            try
            {
                // configure the factory
                gfactory.setRequireTransaction(true);
                gfactory.setAutoStartTx(false);
                gfactory.setWarnOnForceClosingTx(true);
                gfactory.setUseLightweightEdges(true);
                gfactory.setupPool(1, 10);

                // write some records in a transaction
                OrientGraph txgraph = gfactory.getTx();
                try
                {
                    for (int rec = 0; rec < RecordCount; rec++)
                    {
                        boolean tx = gfactory.getDatabase().getTransaction().isActive();
                        HashMap<String, Object> props = new HashMap<>();
                        props.put("name", "vertex-" + rec);
                        props.put("fakeid", rec);
                        txgraph.addVertex("class:V", props);
                    }
                    txgraph.commit();
                }
                catch (Exception ex)
                {
                    System.err.println("write exception! " + ex);
                }
                finally
                {
                    txgraph.shutdown();
                }
            }
            catch (Exception ex)
            {
                System.err.println("test exception! " + ex);
                ex.printStackTrace(System.err);
            }
            finally
            {
                System.out.println("check connection count");
                Thread.sleep(20000);
                gfactory.close();
            }
        }
        catch (Exception ex)
        {
            System.err.println("admin exception! " + ex);
            ex.printStackTrace(System.err);
        }
    }
}
```

**Comment by tglman** [comment]
hi @8forty,

for check that the transaction is active on your current session you should use:
```
txgraph.getRawGraph().getTransaction().isActive()
```
the api:
```
gfactory.getDatabase()
```
open a new session, so if you use it, you should call close the the instance returned by it.

**Comment by 8forty** [comment]
Ok, but then why does it open 1000 connections?  I set the pool size of the factory to 10.

**Comment by tglman** [comment]
Hi @8forty,

The pool size is not the number of connection you do, that is configured by: http://orientdb.com/docs/last/Configuration.html#client-channel-maxpool.

**Comment by tglman** [comment]
hi @8forty,

few issue about connections limit have been solved in the last few hotfix, the 2.1.24 and 2.1.10 should not have anymore this problem, going to close this, pleas reopen if you still see problems.

bye
